packages/core/src/fhir-to-cda/cda-templates/components/__tests__/encounters.test.ts

```diff
@@ -0,0 +1,101 @@
+import { faker } from "@faker-js/faker";
+import { Bundle, Encounter, Practitioner, Location } from "@medplum/fhirtypes";
+import path from "path";
+import _ from "lodash";
+import { removeEmptyFields } from "../../clinical-document/clinical-document";
+import { xmlBuilder } from "../../clinical-document/shared";
+import { buildEncounters } from "../encounters";
+import { encounter1, practitioner1, location1 } from "./encounter-examples";
+import { makeEncounter, makeLocation, makePractitioner } from "./make-encounter";
+import { createEmptyBundle, getXmlContentFromFile } from "./shared";
+
+let encounterId: string;
+let practitionerId: string;
+let locationId: string;
+let bundle: Bundle;
+let encounter: Encounter;
+let practitioner: Practitioner;
+let location: Location;
+
+beforeEach(() => {
+  encounterId = faker.string.uuid();
+  practitionerId = faker.string.uuid();
+  locationId = faker.string.uuid();
+
+  encounter = makeEncounter(
+    { ...encounter1 },
+    {
+      enc: encounterId,
+      pract: practitionerId,
+      loc: locationId,
+    }
+  );
+  practitioner = makePractitioner({ ...practitioner1, id: practitionerId });
+  location = makeLocation({ ...location1, id: locationId });
+
+  bundle = createEmptyBundle();
+});
```

Comment:
Just a few notes on unit test best practices and how Jest/`beforeEach` works:
- We're building a lot of unit tests (which is a good thing!) and the amount will grow over time to a certain plateau - we'll have a lot of unit tests!
- So we want each unit test to be super fast, since it will build up the time needed to run the whole test suite - I've seen this taking longer that the actual deployment 😨 
- `beforeEach` runs before each single test, whereas `beforeAll` runs a single time before all tests on the particular scope (file or `describe`).
- So, obviously, its better _if we can_ add things to `beforeAll` than `beforeEach`, since it will run fewer times and consume less resources/time.
- In this case, we're building a few thing in memory before each unit test, which is fine if we need them for all unit tests;
- But, if some are used in certain unit tests and others are used in diff unit tests, I suggest we create/initialize those on their particular unit test.

I know we're trained to minimize duplicating code and whatnot, but for tests sometimes it's better to duplicate things - https://mtlynch.io/good-developers-bad-tests/ (this is just a thought-provoking read 😄)

Reply:
This makes a lot of sense. Thanks for bringing this up!

Reply:
I'll put the article on my reading list :P